modules/hooks.nix

```diff
@@ -2598,6 +2612,31 @@ lib.escapeShellArgs (lib.concatMap (ext: [ "--ghc-opt" "-X${ext}" ]) hooks.ormol
             builtins.toString script;
           files = "\\.go$";
         };
+      goimports =
+        {
+          name = "goimports";
+          description = "Go import lines, adding missing ones and removing unreferenced ones.";
+          package = tools.gotools;
+          entry =
+            let
+              script = pkgs.writeShellScript "precommit-goimports" ''
+                set -e
+                failed=false
+                for file in "$@"; do
+                    # redirect stderr so that violations and summaries are properly interleaved.
+                    if ! ${hooks.gotools.package}/bin/goimports ${hooks.goimports.settings.flags} -w "$file" 2>&1
```

Comment:
Is there a way to do with args instead of manually specifying flags here and setting them above?

Reply:
Don't think so, since `$@` is already used for the files.

Reply:
If the tool is not too complicated, the general pattern is that we add dedicated options to make them show up in the docs of downstream projects (flake-parts, devenv). 
Some hooks will have an `extraFlags`/`extraOptions` setting to capture everything. Use the existing hooks for inspiration.